### ISSUE TITLE
PHATE for single cell HiC data?

### ISSUE THREAD

**Issue by tarak77** [KrishnaswamyLab/PHATE#34]
Hello,
Compared to population based HiC data, single cell HiC data are very sparse and hence good imputation methods are needed to fill in the contact matrix. 
In line with that, can I use PHATE directly on a very sparse single cell contact matrix? Or will I need to use MAGIC for imputation and then use PHATE on the imputed contact matrix?
Any suggestions will be great!

Also in the age of single cell HiC, it would be awesome if there were some examples of the same in PHATE and/or MAGIC tutorials.
 
Thanks.

**Comment by kmoon3** [comment]
Hi,

PHATE automatically imputes missing data via the process of diffusion. Thus you shouldn't need to run MAGIC prior to running PHATE on the data.

**Comment by scottgigante** [comment]
@tarak77 I might also note that since you are providing a contact matrix, you should flag to PHATE that the distance matrix is precomputed.

In Python:
```
>>> import phate
>>> contact_map = ...
>>> contact_map.shape
(n, n)
>>> phate_operator = phate.PHATE(knn_dist='precomputed')
>>> contact_phate = phate_operator.fit_transform(contact_map)
```

In R:
```
> library(phateR)
> contact_map <- ...
> dim(contact_map)
[1] n n
> contact_phate <- phate(contact_map, knn.dist='precomputed')
```

**Comment by tarak77** [comment]
@scottgigante thanks, will do

**Comment by tarak77** [comment]
@kmoon3 @scottgigante The problem with using binary contact maps with 1 along the diagonals as input to PHATE(using default parameters) is that a compact(kinda globular) structure is obtained. We do not want that! So I think this points to: first impute the missing values(due to very sparse single cell Hi-C data) by using MAGIC before applying PHATE?

I tried to use PHATE on chromosomes for cell 1 and cell 2 from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE80280. At the bottom of that page we use GSE80280_RAW.tar custom file.

In that, the RAW results typically lists a number of observed contacts between specific genome positions. From the raw results, the contacts were aggregated into equally spaced bins along the chromosomes. I worked with a bin size of 50 kbp. Then all observed contacts were assigned to their corresponding bins. In the case that multiple contacts fell into the same bin, the duplicate entries were ignored so that a binary contact matrix C_ij was obtained for each chromosome. Hence, C_ij = 1 represents a Hi-C contact between bins i and j, while C_ij = 0 represents the absence of a contact.

An example file for cell-1 chr1 is below:
The first row list the #bins and #contacts
[GSM2219497_cell1_chr1_50kb.txt](https://github.com/KrishnaswamyLab/PHATE/files/2113117/GSM2219497_cell1_chr1_50kb.txt)

After removing bins with no interactions, the binary contact matrix for the same chr1 is:
[contact_cell1_chr1.txt.tar.gz](https://github.com/KrishnaswamyLab/PHATE/files/2113126/contact_cell1_chr1.txt.tar.gz)

Any help would be greatly appreciated!
Thanks,
-Tarak
